.github/workflows/collector-builder.yml

```diff
@@ -0,0 +1,141 @@
+name: Collector builder image build
+
+on:
+  workflow_call:
+    inputs:
+      collector-tag:
+        type: string
+        required: true
+        description: |
+          The tag used to build the collector image
+    outputs:
+      collector-builder-tag:
+        description: The builder tag used by the build
+        value: ${{ jobs.build-builder-image.outputs.collector-builder-tag }}
+
+env:
+  COLLECTOR_TAG: ${{ inputs.collector-tag }}
+  DEFAULT_BUILDER_TAG: cache
+
+jobs:
+  builder-needs-rebuilding:
+    name: Determine if builder image needs to be built
+    runs-on: ubuntu-latest
+    outputs:
+      build-image: ${{ steps.changed.outputs.builder-changed }}
+
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: changed
+        with:
+          filters: |
+            builder-changed:
+              - builder/**
+              - third_party/**
+
+  build-builder-image:
+    name: Build the builder image
+    runs-on: ubuntu-latest
+    # Multiarch builds sometimes take for eeeeeeeeeever
+    timeout-minutes: 480
+    needs:
+    - builder-needs-rebuilding
+    if: |
+      needs.builder-needs-rebuilding.outputs.build-image == 'true' ||
+      (github.event_name == 'push' && github.ref_type == 'tag') ||
+      contains(github.event.pull_request.labels.*.name, 'build-builder-image')
+    outputs:
+      collector-builder-tag: ${{ steps.builder-tag.outputs.collector-builder-tag }}
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [amd64, ppc64le, s390x]
+
+    env:
+      PLATFORM: linux/${{ matrix.arch }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Define builder tag
+        id: builder-tag
+        run: |
+          COLLECTOR_BUILDER_TAG="${DEFAULT_BUILDER_TAG}"
+          if [[ "${{ github.event_name }}" == 'pull_request' ]]; then
+            COLLECTOR_BUILDER_TAG="${{ inputs.collector-tag }}"
+          fi
+
+          echo "COLLECTOR_BUILDER_TAG=${COLLECTOR_BUILDER_TAG}" >> "$GITHUB_ENV"
+          echo "collector-builder-tag=${COLLECTOR_BUILDER_TAG}" >> "$GITHUB_OUTPUT"
```

Comment:
When the builder does not need to be rebuilt, how is the `collector-build-tag` going to be provided to the collector build ?

Reply:
I'll test it, but when skipping I think the output turns into an empty string and our makefiles turn an empty `COLLECTOR_BUILDER_TAG` env variable to the default (`cache`):
https://github.com/stackrox/collector/blob/d6cd0ff497582e56104911fdb5af966cdbeee077/Makefile-constants.mk#L2-L4

Reply:
Actually, in the latest CI run, that is what happened:
![image](https://github.com/stackrox/collector/assets/20206636/9b14bb73-2e7c-4adc-a285-a98435efaf4c)

https://github.com/stackrox/collector/actions/runs/5856587306/job/15876735578?pr=1284

Reply:
I think that skipping setting of the value when we are in the default case would be coherent (have the default value's "value" in one place). But, hey, it works !